auth-service/src/main/java/org/libertybikes/auth/service/JwtAuth.java

```diff
@@ -0,0 +1,121 @@
+
+package org.libertybikes.auth.service;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.security.Key;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+import java.util.Calendar;
+import java.util.Map;
+
+import javax.annotation.Resource;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+
+/**
+ * A base class for bikes auth implementations that return signed JWTs to the
+ * client.
+ */
+public abstract class JwtAuth {
+
+    private static final long serialVersionUID = 1L;
+
+    @Resource(lookup = "jwtKeyStore")
```

Comment:
to use a consistent config model, lets use `@Inject @ConfigProperty` from MP Config here instead of JNDI lookups

Reply:
ignore this comment. MP Config doesn't have a good way to incorporate ${server.config.dir} yet, so lets stick with JNDI